File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            l, r = 0, len(word) - 1
            pal = True
            while l <= r:
                if word[l] != word[r]:
                    pal = False
                    break
                l += 1
                r -= 1
                
            if pal:
                return word

        return ""

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.cpp
auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    const string firstPalindrome(const vector<string>& words) {
        for (const string &word : words) {
            if (isPalindrome(word)) {
                return word;
            }
        }
        return "";
    }
    const bool isPalindrome(const string &str) {
        size_t l = 0, r = str.size() - 1;
        while (l < r) {
            if (str[l] != str[r]) {
                return false;
            }
            ++l;
            --r;
        }
        return true;
    }
};